src/js/forms.js

```diff
@@ -1,65 +1,51 @@
+import Input from './input';
+import State from './state';
+
 class Forms {
 	/**
 	* Class constructor.
 	* @param {HTMLElement} [formElement] - The form element in the DOM
 	* @param {Object} [options={}] - An options object for configuring the form
 	*/
 	constructor(formElement, options) {
-		this.formEl = formElement;
-		this.inputFields = Array.from(formElement.querySelectorAll('input', 'select', 'textarea'));
+		if (formElement.nodeName !== 'FORM') {
+			throw new Error(`[data-o-component="o-forms"] must be set on a form element. It is currently set on a '${formElement.nodeName.toLowerCase()}'.`);
+		}
+
+		this.form = formElement;
+		this.formElements = Array.from(this.form.elements, element => new Input(element));
+		this.stateArray = [];
 
 		this.opts = Object.assign({
-			useBrowserValidation: true
+			useBrowserValidation: false
 		}, options);
 
-		this.className = {
-			invalid: 'o-forms-input--invalid',
-			valid: 'o-forms-input--valid'
-		};
-
 		if (!this.opts.useBrowserValidation) {
-			this.formEl.setAttribute('novalidate', true);
-			this.formEl.addEventListener('submit', this);
+			this.form.setAttribute('novalidate', true);
+			this.form.addEventListener('submit', this);
 		} else {
-			this.formEl.removeAttribute('novalidate');
-			let submit = this.formEl.querySelector('input[type=submit]');
+			this.form.removeAttribute('novalidate');
+			let submit = this.form.querySelector('[type=submit]');
```

Comment:
This should maybe cater for forms with multiple `submit` inputs. I don't know whether we use them anywhere at the moment, but an example use case would be something like GitHub's "Comment" / "Close and comment". Here, the submit that you click changes the behaviour of the form.

Reply:
It does change the behaviour of the form, but only in so far that it validates the whole form before it submits—it has zero opinions as to what happens on the actual submit.

I'll query for multiple submit inputs/buttons to handle cases where there might be more than one 👍